src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php

```diff
@@ -0,0 +1,252 @@
+     * @param array $options
+     *
+     * @return MemcachedAdapter
+     */
+    public static function createAdapter(array $servers = array(), array $options = array())
+    {
+        $adapter = new static(static::createClient());
+        $adapter->setupConnection($servers, $options);
+
+        return $adapter;
+    }
+
+    /**
+     * @param array $servers
+     * @param array $options
+     *
+     * @return \Memcached
+     */
+    public function setupConnection(array $servers = array(), array $options = array())
+    {
+        foreach ($servers as $server) {
+            $this->addServer($server);
+        }
+
+        foreach ($options as $opt => $val) {
+            $this->setOption($opt, $val);
+        }
+
+        return $this->memcached;
+    }
+
+    /**
+     * @param mixed[] $server
+     */
+    public function addServer(array $server)
+    {
+        list($host, $port, $weight) = $this->filterServerValues(
+            isset($server[0]) ? $server[0] : self::DEFAULT_HOST,
+            isset($server[1]) ? $server[1] : self::DEFAULT_PORT,
+            isset($server[2]) ? $server[2] : self::DEFAULT_WEIGHT
+        );
+
+        if (!$this->isServerAlreadyEnabled($host, $port)) {
+            $this->memcached->addServer($host, $port, $weight);
+        }
+    }
+
+    /**
+     * @param mixed $opt
+     * @param mixed $val
+     *
+     * @return bool
+     */
+    public function setOption($opt, $val)
+    {
+        list($opt, $val) = $this->filterOptionValues($opt, $val);
+
+        return $this->memcached->setOption($opt, $val)
+            && $this->memcached->getResultCode() === \Memcached::RES_SUCCESS;
+    }
+
+    /**
+     * @param mixed $opt
+     * @param mixed $val
+     *
+     * @return mixed[]
+     */
+    private function filterOptionValues($opt, $val)
+    {
+        list($opt, $val) = array(
+            $opt = $this->tryResolvingConstant($opt),
+            $val = $this->tryResolvingConstant($val),
+        );
+
+        if (false === filter_var($opt, FILTER_VALIDATE_INT)) {
+            throw new InvalidArgumentException(sprintf('Option type invalid "%s": must resolve as int', $opt));
+        }
+
+        if (false === filter_var($val, FILTER_VALIDATE_INT) && false === filter_var($val, FILTER_VALIDATE_BOOLEAN)) {
+            throw new InvalidArgumentException(sprintf('Option value invalid "%s": must resolve as int or bool', $val));
+        }
+
+        return array($opt, $val);
+    }
+
+    /**
+     * @param mixed $val
+     *
+     * @return mixed
+     */
+    private function tryResolvingConstant($val)
+    {
+        return defined($constant = '\Memcached::'.$val) ? constant($constant) : $val;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
+    {
+        foreach ($this->memcached->getMulti($ids) as $id => $val) {
+            try {
+                yield $id => parent::unserialize($val);
+            } catch (\Exception $e) {
```

Comment:
Left over from Redis basis for this class. Gone at this point. Also, re: serializtion: I assumed there was a specific requirement to handle serialization using the AbstractAdapter implementation (I guess Redis doesn't handle this for you?). Anyway, the double serialization has been removed at this point. Will push the latest code changes shortly.